pkg/operator/controller/ingress/status.go

```diff
@@ -204,18 +203,17 @@ func computeAllowedSourceRanges(service *corev1.Service) []operatorv1.CIDR {
 	return nil
 }
 
-// computeDeploymentPodsScheduledCondition computes the ingress controller's
-// current PodsScheduled status condition state by inspecting the PodScheduled
-// conditions of the pods associated with the deployment.
-func computeDeploymentPodsScheduledCondition(deployment *appsv1.Deployment, pods []corev1.Pod) operatorv1.OperatorCondition {
+// checkPodsScheduledForDeployment checks whether the given deployment has a
+// valid label selector and whether all of the deployment's pods are reporting
+// PodsScheduled=True.  This function returns an error value indicating the
+// result of that check.
+func checkPodsScheduledForDeployment(deployment *appsv1.Deployment, pods []corev1.Pod) error {
+	if deployment == nil {
+		return errors.New("System error detected: deployment was nil.  Please report this issue to Red Hat: https://issues.redhat.com/secure/CreateIssueDetails!init.jspa?pid=12332330&issuetype=1&components=12367900&priority=10300&customfield_12316142=26752")
```

Comment:
I have some concern this link may break in the future. Would it be better to just do `https://issues.redhat.com/`?

Reply:
I think that that would lead to more misrouted bug reports.  

I could say "open a support case", but I don't even know the link for that, and neither would OKD users.

Reply:
:man_shrugging: this is fine, hopefully we never hit this logic